types/index.d.ts

```diff
@@ -60,10 +60,10 @@ interface AsyncStorage {
 }
 
 type AsyncStorageHook = {
-  getItem(callback?: (error?: Error, result?: string) => void): Promise<string | null>;
-  setItem(value: string, callback?: (error?: Error) => void): Promise<void>;
-  mergeItem(value: string, callback?: (error?: Error) => void): Promise<void>;
-  removeItem(callback?: (error?: Error) => void): Promise<void>;
+  getItem(this: void, callback?: (error?: Error, result?: string) => void): Promise<string | null>;
+  setItem(this: void, value: string, callback?: (error?: Error) => void): Promise<void>;
+  mergeItem(this: void, value: string, callback?: (error?: Error) => void): Promise<void>;
+  removeItem(this: void, callback?: (error?: Error) => void): Promise<void>;
```

Comment:
According to the warning, we can also use arrow function. I think this is more legible:
```suggestion
  getItem: (callback?: (error?: Error, result?: string) => void) => Promise<string | null>;
  setItem: (value: string, callback?: (error?: Error) => void) => Promise<void>;
  mergeItem: (value: string, callback?: (error?: Error) => void) => Promise<void>;
  removeItem: (callback?: (error?: Error) => void) => Promise<void>;
```

Reply:
Yeah, that's more familiar to users who don't know TS but do know JS as well.